src/orion/core/io/cmdline_parser.py

```diff
@@ -62,14 +73,15 @@ def parse(self, commandline):
                     self.template.append(template)
                     continue
 
-                for pos, item in enumerate(value):
+                for pos in range(len(value)):
                     template = "{" + key + "[" + str(pos) + "]}"
                     self.template.append(template)
 
         self._already_parsed = True
 
         return self.arguments
 
+    # pylint: disable=no-self-use
     def _key_to_arg(self, key):
         arg = key.replace("!!", "_").replace("??", "-")
```

Comment:
Those magic values `"!!"` for `"_"`  and `"??"` for `"-" should be made attributes so that they are customizable.